realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -633,15 +633,16 @@ private void checkEmpty(String str) {
             }
             return columnIndices;
         } else {
-            if (getFieldIndex(fieldDescription) == null) {
+            Long fieldIndex = getFieldIndex(fieldDescription);
+            if (fieldIndex == null) {
                 throw new IllegalArgumentException(String.format("Field '%s' does not exist.", fieldDescription));
```

Comment:
uh... we might need to create a task to clean up exception messages ... OK to make it like this in this PR?